src/test/kotlin/com/parmet/buf/gradle/AbstractGenerateTest.kt

```diff
@@ -38,4 +38,21 @@ abstract class AbstractGenerateTest : AbstractBufIntegrationTest() {
             listOf("build", "bufbuild", "generated", "java", "com", "google", "type", "DateTime.java")
         assertThat(Paths.get(projectDir.absolutePath, *generatedPathElements.toTypedArray()).toFile().exists()).isTrue()
     }
+
+    @Test
+    fun `generate java with gen file override`() {
+        gradleRunner().withArguments(BUILD_TASK_NAME).build()
+    }
+
+    @Test
+    fun `generate java with gen file override fails with no files`() {
+        val result = gradleRunner().withArguments(BUILD_TASK_NAME).buildAndFail()
+        assertThat(result.output).contains("No buf.gen.yaml file found in the root directory or with genFileLocation")
+    }
+
+    @Test
+    fun `generate java with gen file override fails with two files`() {
```

Comment:
And: `buf generate fails with both default and specified buf gen template files`.